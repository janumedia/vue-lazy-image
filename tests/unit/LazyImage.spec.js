import { shallowMount } from '@vue/test-utils'
import LazyImage from '@/LazyImage'

const createWrapper = propsData => shallowMount(LazyImage, {
    attachToDocument: true,
    propsData
});

describe('LazyImage.vue', () => {
    let wrapper,
        IMAGE               = 'image.jpg',
        IMAGE_PLACEHOLDER   = 'custom-image-placeholder.jpg',
        props               = { src: IMAGE, alt: 'image title'};
    
    beforeAll(() => {
        LazyImage.components = LazyImage.components || {};
    });

    afterEach(() => {
        //destory wrapper
        wrapper.destroy();
    });

    it('should render LazyImage default', () => {
        wrapper = createWrapper(props);
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render LazyImage custom placeHolder', () => {
        wrapper = createWrapper({...props, placeHolder: IMAGE_PLACEHOLDER});
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render LazyImage custom width', () => {
        wrapper = createWrapper({...props, width: 300});
        expect(wrapper.html()).toMatchSnapshot();
    });
})