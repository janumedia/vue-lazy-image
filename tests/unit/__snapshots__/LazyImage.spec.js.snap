// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LazyImage.vue should render LazyImage custom placeHolder 1`] = `<div class="lazy-image"><img src="custom-image-placeholder.jpg" alt="image title" data-src="image.jpg" data-pin-nopin="true"></div>`;

exports[`LazyImage.vue should render LazyImage custom width 1`] = `<div class="lazy-image" style="width: 300px;"><img src="assets/image-placeholder.svg" alt="image title" data-src="image.jpg" data-pin-nopin="true"></div>`;

exports[`LazyImage.vue should render LazyImage default 1`] = `<div class="lazy-image"><img src="assets/image-placeholder.svg" alt="image title" data-src="image.jpg" data-pin-nopin="true"></div>`;
